### PR TITLE
feat(frontend): Add flag to mock API during dev mode

### PR DESCRIPTION
**Pull request by amanape**
**End-user friendly description of the problem this fixes or functionality that this introduces**
- `npm run dev` does NOT mock the API
- `npm run dev:mock` does mock the API

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

---
**Link of any specific issues this addresses**

### CODE REVIEW

**Comment by tobitege** [comment]
`npm run dev:mock` could be easier to remember than `npm run dev:only`?

**Comment by amanape** [comment]
Yes but I want to keep the default as mocking the API. `npm run dev:mock` is a good alternative but that would mean the default is NOT mocking the API.

Maybe it would be better to have mocking secondary 😢

**Comment by tobitege** [comment]
Ohh, gotcha, nevermind then. :)

File: frontend/package.json [hunk 1]
```diff
@@ -43,7 +43,8 @@
     "ws": "^8.18.0"
   },
   "scripts": {
-    "dev": "npm run make-i18n && remix vite:dev",
+    "dev": "npm run make-i18n && VITE_MOCK_API=true remix vite:dev",
+    "dev:only": "npm run make-i18n && VITE_MOCK_API=false remix vite:dev",
```
Comment (amanape) [inline frontend/package.json:47]:
```suggestion
    "dev": "npm run make-i18n && VITE_MOCK_API=false remix vite:dev",
    "dev:mock": "npm run make-i18n && VITE_MOCK_API=true remix vite:dev",
```

File: frontend/.env.sample [hunk 2]
```diff
@@ -1,4 +1,5 @@
 VITE_BACKEND_BASE_URL="localhost:3000" # Backend URL without protocol (e.g. localhost:3000)
+VITE_MOCK_API="true" # true or false
```
Comment (amanape) [inline frontend/.env.sample:2]:
```suggestion
VITE_MOCK_API="false" # true or false
```

**Review by tofarr — APPROVED** [review]
🍰

**Pull request merged**